SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1072,30 +1072,65 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 	});
 }
 
-void USpatialActorChannel::ServerProcessOwnershipChange()
+void USpatialActorChannel::StartServerProcessOwnershipChange()
 {
 	if (!IsAuthoritativeServer())
 	{
 		return;
 	}
 
-	UpdateEntityACLToNewOwner();
+	// If you attempt to take possession of an actor, the server will first take authority over the ClientRPC component.
+	// It will later give authority to the requested client worker.
+	// If the actor is being unpossessed (NewOwnerWorkerAttribute is empty) we simply remove the client workers authority over the ClientRPC component. We do no hand authority to the server.
+
+	FString NewOwnerWorkerAttribute = SpatialGDK::GetOwnerWorkerAttribute(Actor);
+	if (!NewOwnerWorkerAttribute.IsEmpty())
+	{
+		// Setting CachedOwnerWorkerAttribute will trigger FinishServerProcessOwnershipChange from HandleAuthority once authority has been confirmed.
+		CachedOwnerWorkerAttribute = NewOwnerWorkerAttribute;
+		NewOwnerWorkerAttribute = SpatialConstants::DefaultServerWorkerType.ToString();
+	}
+	UpdateEntityACLToNewOwner(NewOwnerWorkerAttribute);
 
 	for (AActor* Child : Actor->Children)
 	{
 		Worker_EntityId ChildEntityId = NetDriver->PackageMap->GetEntityIdFromObject(Child);
 
 		if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(ChildEntityId))
 		{
-			Channel->ServerProcessOwnershipChange();
+			Channel->StartServerProcessOwnershipChange();
 		}
 	}
 }
 
-void USpatialActorChannel::UpdateEntityACLToNewOwner()
+void USpatialActorChannel::FinishServerProcessOwnershipChange()
```

Comment:
Could you add a comment that this is called from `USpatialReciever::HandleActorAuthority`